napari/viewer.py

```diff
@@ -19,6 +22,29 @@ class Viewer(ViewerModel):
     """
 
     def __init__(self, title='napari', ndisplay=2, order=None):
+        # instance() returns the singleton instance if it exists, or None
+        app = QApplication.instance()
+        # if None, wait two seconds, as it may still be instantiating when
+        # using %gui qt inside IPython
+        if app is None:
+            sleep(2)
```

Comment:
This trick with `sleep(2)` is not working for me if I have one jupyter cell with both the magic and the view_image command in it. Even if I make it `sleep(20)` it still doesn't work, so I guess it is blocking the construction of the event loop. Similarly when launching from ipython I couldn't find a condition where having two checks and the sleep actually helped anything. I think we should remove the sleep and just raise the error - which is SO much better than the kernel dying like it used to. Unless you were able to find a condition when having the sleep in there prevented an error.